lib/cocoapods/target/pod_target.rb

```diff
@@ -117,15 +117,17 @@ def scoped(cache = {})
         if cache[cache_key]
           cache[cache_key]
         else
-          target = PodTarget.new(sandbox, host_requires_frameworks, user_build_configurations, archs, platform, specs, [target_definition], file_accessors, target_definition.label,
-                                 :build_type => build_type)
-          target.dependent_targets = dependent_targets.flat_map { |pt| pt.scoped(cache) }.select { |pt| pt.target_definitions == [target_definition] }
-          target.test_dependent_targets_by_spec_name = Hash[test_dependent_targets_by_spec_name.map do |spec_name, test_pod_targets|
-            scoped_test_pod_targets = test_pod_targets.flat_map do |test_pod_target|
-              test_pod_target.scoped(cache).select { |pt| pt.target_definitions == [target_definition] }
-            end
-            [spec_name, scoped_test_pod_targets]
-          end]
+          target = PodTarget.new(sandbox, host_requires_frameworks, user_build_configurations, archs, platform, specs,
+                                 [target_definition], file_accessors, target_definition.label, :build_type => build_type)
+          target.dependent_targets = dependent_targets.flat_map { |pt| pt.scoped(cache) }.select do |pt|
+            pt.target_definitions == [target_definition]
```

Comment:
super nit: this creates a new array in the loop, pulling it out would save some memory